src/mds/CInode.cc

```diff
@@ -4237,10 +4242,16 @@ void CInode::validate_disk_state(CInode::validated_data *results,
       if (divergent || memory_newer < 0) {
         // we're divergent, or on-disk version is newer
         results->backtrace.error_str << "On-disk backtrace is divergent or newer";
-        // we probably have a new unwritten file!
-        // so skip the backtrace scrub for this entry and say that all's well
-        if (divergent && in->is_dirty_parent())
+        /* if the backtraces are divergent and the link count is 0, then
+         * most likely its a stray entry that's being purged and things are
+         * well and there's no reason for alarm
+         */
+        if (divergent && (in->is_dirty_parent() || in->get_inode()->nlink == 0)) {
```

Comment:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/CInode.cc: In member function 'bool CInode::validate_disk_state(CInode::validated_data*, MDSContext*)::ValidationContinuation::_dirfrags(int)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/CInode.cc:52:88: error: invalid use of non-static data member 'InodeStoreBase::inode'

Reply:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/CInode.cc:4249:67: error: base operand of '->' has non-pointer type 'InodeStoreBase::mempool_inode' {aka 'inode_t<mempool::mds_co::pool_allocator>'}
 4249 |         if (divergent && (in->is_dirty_parent() || in->get_inode()->nlink == 0)) {
      |                                                                   ^~
```